src/corecel/sys/KernelDiagnostics.cc

```diff
@@ -26,15 +40,15 @@ namespace celeritas
  */
 void KernelDiagnostics::log_launch(value_type& diag, unsigned int num_threads)
 {
-    static unsigned int remaining_log_msg = 32;
-    if (remaining_log_msg == 0)
-        return;
-
-    CELER_LOG(debug) << "Launching '" << diag.name << "' on "
-                     << diag.threads_per_block << " threads per block with "
-                     << num_threads << " total threads";
-
-    if (--remaining_log_msg == 0)
+    static std::atomic<int> remaining_log_msg{32};
+    int                     remaining = --remaining_log_msg;
+    if (remaining > 0)
```

Comment:
```suggestion
    if (remaining < 0) {
       remaining = 0;
    } else if (remaining > 0)
```
otherwise, the logging may restart if there is really a log of errors (`INT_MAX`).

Reply:
Ah, I had that in there, but then removed it because (1) I forgot that since I'm using the prefix decrement resetting it to zero will *not* show the "ignoring" message on subsequent calls (2) I figured we'd never launch 2 billion kernels per execution?

Reply:
> (2) I figured we'd never launch 2 billion kernels per execution?

Since we are launching several kernels per steps and they are pretty short lived and we could be using several cards and processes could run for several days (for 7 days, 4 cards, if I am not mistaken it would take "only" 900 kernels a seconds per cards).   So hard, not any time soon but possible.

Reply:
Fair enough!